### DISCUSSION TITLE
X1-Micro microinverter range

### DISCUSSION THREAD

**Discussion by raddzz** [squishykid/solax#182] [General]
Solax has recently announced microinverters, I've picked up a few and am hoping to start seeing how they communicate.

The software they run seems to be different from other solax equipment, they do allow local access when connected to the home wifi network, but from what I've been able to find so far this is just for configuration and not for generation data.

As of yet I've been unable to get any data for them from SolaxCloud, whenever I try I'm told that the token is invalid, but I think it's because the API isn't setup to recognise the serial numbers of microinverters yet.

This is the request which is made once logged in to the http interface, I've not yet been able to find any other endpoints that return any data.
![image](https://github.com/user-attachments/assets/34651828-9711-42f1-982b-0663c39ce75d)

Just wondering if anyone else has looked into this, or recognises the interface.
![image](https://github.com/user-attachments/assets/64eca806-0ff8-4815-b3da-bdae49098392)

**Comment by gruntmar** [comment]
I have captured some local communication between the solaxcloud app and X1-Micro 2 in 1 inverter.

It starts with simple query that i was able to replicate with python. All further communication base64 encoded.
After encoding it is partly ASCII  and partly position encoded data. Could not extract any valuable info so far.
Maybe it helps someone to get further.

```
import requests

url = "http://192.168.10.10/"
headers = {
    "Content-Type": "application/x-www-form-urlencoded",
    "User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 18_4_1 like Mac OS X)",
}
data = "optType=getModuleInfo"
response = requests.post(url, headers=headers, data=data)
print(response.text)

```
**Response:**
HTTP/1.1 200 OK
Content-Type: text/html
Content-Length: 169

 {"code":200,"message":"success","data":{"version":"2.0","encryption":0,"category":2,"deviceType":14,"inverters":[{"sn":"XXXXXXXXXXXX","category":1,"deviceType":28}]}}
 
 I'm wondering what other option types might be available to request possible usable data.
